src/language-js/printer-estree.js

```diff
@@ -4075,14 +4075,30 @@ function printArgumentsList(path, options, print) {
       arg &&
       arg.type === "ArrowFunctionExpression" &&
       arg.params &&
-      arg.params.some(hasEmptyLineInObject)
+      arg.params.some(
+        param =>
+          (param.type === "ObjectPattern" && hasEmptyLineInObject(param)) ||
+          (param.type === "ArrayPattern" &&
+            param.elements &&
+            param.elements.some(
+              element =>
+                (element.type === "ObjectPattern" &&
+                  hasEmptyLineInObject(element)) ||
+                (element.type === "RestElement" &&
+                  element.argument &&
+                  hasEmptyLineInObject(element.argument))
+            )) ||
+          (param.type === "AssignmentPattern" &&
+            param.right &&
+            hasEmptyLineInObject(param.right))
+      )
```

Comment:
maybe we can move this to function, for example hasNewlineInParams?

Reply:
Where should I move the function? Also, [this parts](https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L4092-L4112) is related?

Reply:
just question :smile:

Reply:
Oh, sorry.
I might be able to move it. But, I think I should not do it in this pr.
Probably, the range of influence will be larger.